# City Exploration from Inside Airbnb Data

This notebook analyzes the scraped Inside Airbnb data to explore cities and their data availability.

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('../insideairbnb_all_data.csv')
print(f"Loaded {len(df)} rows of data.")
df.head()

Loaded 506 rows of data.


City    Region        Country        Date  \
0  Albany  New York  United States  2025-09-06   
1  Albany  New York  United States  2025-08-04   
2  Albany  New York  United States  2025-07-04   
3  Albany  New York  United States  2025-06-09   
4  Albany  New York  United States  2025-05-02   

                                        listings_url  \
0  https://data.insideairbnb.com/united-states/ny...   
1  https://data.insideairbnb.com/united-states/ny...   
2  https://data.insideairbnb.com/united-states/ny...   
3  https://data.insideairbnb.com/united-states/ny...   
4  https://data.insideairbnb.com/united-states/ny...   

                                         geojson_url  
0  https://data.insideairbnb.com/united-states/ny...  
1  https://data.insideairbnb.com/united-states/ny...  
2  https://data.insideairbnb.com/united-states/ny...  
3  https://data.insideairbnb.com/united-states/ny...  
4  https://data.insideairbnb.com/united-states/ny...

In [3]:
df['Country'].value_counts()

Country
United States      141
Canada              55
Australia           55
Italy               32
Spain               28
Switzerland         18
Greece              13
France              13
The Netherlands     12
United Kingdom      12
Hungary             10
Belgium              9
Germany              8
Portugal             8
Argentina            4
South Africa         4
China                4
Singapore            4
Taiwan               3
Czech Republic       3
Sweden               3
Japan                3
Brazil               3
Latvia               3
Denmark              3
Norway               3
Mexico               3
Turkey               3
Ireland              3
Belize               3
Thailand             3
Austria              3
Chile                1
Name: count, dtype: int64

In [4]:
df = df[df['Country'].isin(['United States', 'Canada', 'Australia', 'United Kingdom'])]

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 263 entries, 0 to 484
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   City          263 non-null    object
 1   Region        263 non-null    object
 2   Country       263 non-null    object
 3   Date          263 non-null    object
 4   listings_url  263 non-null    object
 5   geojson_url   263 non-null    object
dtypes: object(6)
memory usage: 14.4+ KB


In [7]:
country_dict = {country: df[df['Country'] == country] for country in ['United States', 'Canada', 'Australia', 'United Kingdom']}

In [8]:
for country, country_df in country_dict.items():
    print(f"Country {country}: {len(country_df)} listings")
    print()
    print(country_df['City'].value_counts())
    print("="*20)


Country United States: 141 listings

City
New York City         12
Albany                11
Dallas                11
Bozeman               10
Rochester             10
Los Angeles            4
Santa Clara County     4
San Mateo County       4
San Francisco          4
Portland               4
New Orleans            4
Nashville              4
Boston                 4
Cambridge              4
Hawaii                 3
Rhode Island           3
Seattle                3
Santa Cruz County      3
Austin                 3
Broward County         3
San Diego              3
Chicago                3
Jersey City            3
Pacific Grove          3
Oakland                3
Newark                 3
Columbus               3
Denver                 3
Asheville              3
Fort Worth             3
Twin Cities MSA        3
Name: count, dtype: int64
Country Canada: 55 listings

City
Toronto          11
Vancouver        11
Victoria         11
Quebec City      10
Montreal          3
New Brunswick     3
Ott

In [ ]:
cities_to_explore = [
    "New York City",
    "

Country United States: 141 listings

City

New York City         12
Dallas                11
Los Angeles            4
San Francisco          4
Portland               4
New Orleans            4
Boston                 4

Toronto          11
Vancouver        11
Victoria         11

Brisbane                11
Sunshine Coast          11
Barossa Valley           4
Tasmania                 4